core/lib/Cardano/Address/Script.hs

```diff
@@ -252,25 +372,59 @@ keyHashFromBytes (cred, bytes)
     | otherwise = Just $ KeyHash cred bytes
 
 -- | Encode a 'KeyHash' to bech32 'Text' or hex is key role unknown.
+--  If one wants to include, valid in governance roles only, additional byte
+--  as specified in CIP-0129, the function needs to be called with withByte=true.
 --
 -- @since 3.0.0
-keyHashToText :: KeyHash -> Text
-keyHashToText (KeyHash cred keyHash) = case cred of
+keyHashToText :: KeyHash -> Bool -> Text
```

Comment:
Same remark as above, which makes me think we need something more abstract like

```
data WithByte a = WithByte a | Bare a 
```

Reply:
I will address that in forthcoming refactoring as part of https://github.com/IntersectMBO/cardano-addresses/issues/289